Sources/Xcodeproj/generate().swift

```diff
@@ -77,10 +79,32 @@ public func generate(
     // references in the project.
     let extraDirs = try findDirectoryReferences(path: srcroot)
 
+    // Find non-source files in the source directories and root that should be added
+    // as a reference to the project.
+    var extraFiles = try findNonSourceFiles(path: srcroot)
+
+    let sourcesDirectory = srcroot.appending(component: "Sources")
+    if localFileSystem.isDirectory(sourcesDirectory) {
+        let sourcesExtraFiles = try findNonSourceFiles(path: sourcesDirectory, recursively: true)
+        extraFiles.append(contentsOf: sourcesExtraFiles)
+    }
+
+    let testsDirectory = srcroot.appending(component: "Tests")
+    if localFileSystem.isDirectory(testsDirectory) {
+        let testsExtraFiles = try findNonSourceFiles(path: testsDirectory, recursively: true)
+        extraFiles.append(contentsOf: testsExtraFiles)
+    }
+
+    if try repositoryProvider.checkoutExists(at: srcroot) {
+        // Get the ignored files and exclude them
+        let repository = try repositoryProvider.openCheckout(at: srcroot)
+        extraFiles = try extraFiles.filter { try !repository.isIgnored($0) }
```

Comment:
Minor nitpick: Can you surround the closure with parentheses to match the informal project style? For reference, it follows the Rule of Kevin (“When a trailing closure argument is functional, use parentheses. When it is procedural, use braces.”)